homeassistant/components/alidns/__init__.py

```diff
@@ -0,0 +1,145 @@
+"""Integrate with AliDNS."""
+import asyncio
+from datetime import timedelta
+import json
+import logging
+from re import search
+
+from aliyunsdkalidns.request.v20150109.AddDomainRecordRequest import (
+    AddDomainRecordRequest,
+)
+from aliyunsdkalidns.request.v20150109.DescribeSubDomainRecordsRequest import (
+    DescribeSubDomainRecordsRequest,
+)
+from aliyunsdkalidns.request.v20150109.UpdateDomainRecordRequest import (
+    UpdateDomainRecordRequest,
+)
+from aliyunsdkcore.acs_exception.exceptions import ClientException, ServerException
+from aliyunsdkcore.client import AcsClient
+import async_timeout
+import voluptuous as vol
+
+from homeassistant.const import CONF_DOMAIN, CONF_SCAN_INTERVAL
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+TIMEOUT = 30  # seconds
+INTERNET_IP_URL = "http://www.3322.org/dyndns/getip"
+
+DEFAULT_INTERVAL = timedelta(minutes=10)
+DOMAIN = "alidns"
+
+CONF_ACCESS_ID = "access_id"
+CONF_ACCESS_KEY = "access_key"
+CONF_SUB_DOMAIN = "sub_domain"
+
+CONFIG_SCHEMA = vol.Schema(
```

Comment:
We no longer accept additions or changes to the yaml config of integrations. See: https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md

This integration needs to be refactored to use a config flow and config entries.

https://developers.home-assistant.io/docs/config_entries_config_flow_handler